clang/include/clang/AST/StmtCXX.h

```diff
@@ -524,6 +525,354 @@ class CoreturnStmt : public Stmt {
   }
 };
 
+/// CXXExpansionStmt - Base class for an unexpanded C++ expansion statement.
+class CXXExpansionStmt : public Stmt {
+  friend class ASTStmtReader;
+
+  ExpansionStmtDecl *ParentDecl;
+  SourceLocation ForLoc;
```

Comment:
When is this not just `BeginLoc`?

Reply:
So `getBeginLoc()` for a `CXXExpansionStmt` currently returns the location of the `template` keyword because I figured that made more sense.

Reply:
> What do we use the `For` loc for then?

Nothing really; I just included it because we typically include keyword locations in the AST; I guess it’s just there in case someone wants to do something w/ it?

Reply:
In case we really need it, we could get from it through the source manager - there cannot be anything else here